tests/config_type_test.go

```diff
@@ -0,0 +1,203 @@
+		ctx.testContextChanged(t)
+		now := time.Now()
+		configType := ctx.AdminSession.requireCreateNewConfigType()
+
+		entityJson := ctx.AdminSession.validateEntityWithQuery(configType)
+		ctx.validateDateFieldsForCreate(now, entityJson)
+
+		entityJson = ctx.AdminSession.validateEntityWithLookup(configType)
+		ctx.validateDateFieldsForCreate(now, entityJson)
+	})
+
+	t.Run("create config type with invalid schema should fail", func(t *testing.T) {
+		ctx.testContextChanged(t)
+		configType := ctx.newConfigType()
+		configType.schema = map[string]interface{}{
+			"$id":                  "http://ziti-edge.netfoundry.io/schemas/test.config.json",
+			"type":                 "objectionable", // not a valid type
+			"additionalProperties": false,
+			"required": []interface{}{
+				"hostname",
+				"port",
+			},
+			"properties": map[string]interface{}{
+				"hostname": map[string]interface{}{
+					"type": "string",
+				},
+			},
+		}
+		httpCode, body := ctx.AdminSession.createEntity(configType)
+		ctx.requireFieldError(httpCode, body, apierror.InvalidFieldCode, "schema")
+	})
+
+	t.Run("create config type with schema should pass", func(t *testing.T) {
+		ctx.testContextChanged(t)
+		now := time.Now()
+		configType := ctx.newConfigType()
+		configType.schema = map[string]interface{}{
+			"$id":                  "http://ziti-edge.netfoundry.io/schemas/test.config.json",
+			"type":                 "object",
+			"additionalProperties": false,
+			"required": []interface{}{
+				"hostname",
+				"port",
+			},
+			"properties": map[string]interface{}{
+				"hostname": map[string]interface{}{
+					"type": "string",
+				},
+				"port": map[string]interface{}{
+					"type":    "integer",
+					"minimum": float64(0),
+					"maximum": float64(math.MaxUint16),
+				},
+			},
+		}
+		configType.id = ctx.AdminSession.requireCreateEntity(configType)
+
+		entityJson := ctx.AdminSession.validateEntityWithQuery(configType)
+		ctx.validateDateFieldsForCreate(now, entityJson)
+
+		entityJson = ctx.AdminSession.validateEntityWithLookup(configType)
+		ctx.validateDateFieldsForCreate(now, entityJson)
+	})
+
+	t.Run("lookup non-existent config type as admin should fail", func(t *testing.T) {
+		ctx.testContextChanged(t)
+		ctx.requireNotFoundError(ctx.AdminSession.query("config-types/" + uuid.New().String()))
+	})
+
+	t.Run("lookup config type as non-admin should fail", func(t *testing.T) {
+		ctx.testContextChanged(t)
+		configType := ctx.AdminSession.requireCreateNewConfigType()
+		ctx.requireUnauthorizedError(nonAdminUser.query("config-types/" + configType.id))
+	})
+
+	t.Run("update config type should pass", func(t *testing.T) {
+		ctx.testContextChanged(t)
+		now := time.Now()
+		configType := ctx.AdminSession.requireCreateNewConfigType()
+
+		entityJson := ctx.AdminSession.validateEntityWithQuery(configType)
+		createdAt := ctx.validateDateFieldsForCreate(now, entityJson)
+
+		time.Sleep(time.Millisecond * 10)
```

Comment:
What behaviour were you seeing that required `time.Sleep()`?

Reply:
the validateDateFieldsForUpdate below checks that the updateAt field has been updated since the create above, so it needs a small gap to ensure that it actually does increase.